### ISSUE TITLE
Enum Generating as many times as used in the class

### ISSUE THREAD

**Issue by satyanarayan18** [RSuter/NSwag#1499]
I have created an enum in c#

public enum Color
{
Red,
Green
}

public class A
{
public Color Color
}

public class B
{
public Color Color
}

Enum generated in the client side proxy using n-swag as

export enum AColor
{ 
 _0=0,
_1=1
}

public enum BColor
{
 _0=0,
_1=1
}

Expected result,

export enum Color
{
_0=0,
_1=1
}

**Comment by RSuter** [comment]
How did you generate the Swagger spec? It's probably not generated with NSwag and contains only 0, 1 as enum values (and no names). This is why NSwag cannot know the actual enum value names...

**Comment by RSuter** [comment]
NSwag uses $refs for enums so that they are generated only once, Swashbuckle and a lot of other tools repeat the enums which lead to lots of duplicated code... There is an option issue to first "merge" enums: https://github.com/RSuter/NJsonSchema/issues/17

**Comment by satyanarayan18** [comment]
Thanks for the response. Could you please provide or guide me through some examples to use the Merge enum while generating client side using NSwag

**Comment by RSuter** [comment]
This feature is not implemented yet

**Comment by Estar1** [comment]
Please when will this feature be implemented or any temporary workaround?

**Comment by Kordonme** [comment]
Anything we can do to avoid these duplicate enums in TypeScript?

**Comment by coder8keyboard** [comment]
I'm also seeing nSwag generating multiple classes for one c# enum class. Is it a bug in nSwag?

**Comment by RSuter** [comment]
Duplicate of https://github.com/RSuter/NJsonSchema/issues/17

**Comment by xprees** [comment]
> I've created a regex to remove these duplicate enums using a PowerShell script that is running after NSwag.
> 
> ```powershell
> (Get-Content -Raw ./src/apiClient/apiClient.ts) `
>   -replace '(\w+\??: )([A-Z][A-Za-z]+)(\d+)', '$1$2' `
>   -replace '((\/\*\*[\w\s]+\*\/\s+)?(export enum [A-Za-z]+(\d+)[^}]+})(\s*))', '' |
>   Out-File ./src/apiClient/apiClient.ts -encoding utf8
> ```

In case someone can find this helpful.
I've created a JS script inspired by @Kordonme. I also added it as a script to package.json to simplify the workflow.
Usage: `node fix-api-enums.js ./src/api/api.service.ts`

```javascript
// Usage: node fix-api-enums.js ./src/api/api.service.ts
const fs = require('fs');

const file = process.argv[2];
if (!file || !fs.existsSync(file)) {
    console.error('Please provide a correct file path as an argument.');
    process.exit(1);
}

const content = fs.readFileSync(file, 'utf8');
const updatedContent = content.replace(/(\w+\??: )([A-Z][A-Za-z]+)(\d+)/g, '$1$2');
const finalContent = updatedContent.replace(/(\/\*\*[\w\s]+\*\/\s+)?(export enum [A-Za-z]+(\d+)[^}]+})(\s*)/g, '');
fs.writeFileSync(file, finalContent, 'utf8');

```
